hibike/devices/ExampleDevice/example_device.cpp

```diff
@@ -1,69 +1,116 @@
 #include "example_device.h"
 
 // each device is responsible for keeping track of it's own params
-uint32_t params[NUM_PARAMS];
-
-
 uint8_t value0;
 uint8_t value1;
 uint16_t value2;
 uint32_t value3;
 uint64_t value4;
+uint64_t value5;
+
 
 // normal arduino setup function, you must call hibike_setup() here
 void setup() {
   hibike_setup();
 }
 
+
 // normal arduino loop function, you must call hibike_loop() here
 // hibike_loop will look for any packets in the serial buffer and handle them
-void loop() {
 
+void loop() {
   // do whatever you want here
   // note that hibike will only process one packet per call to hibike_loop()
   // so exessive delays here will affect hibike.
-  value1++;
-  value2++;
-  value3++;
-  value4++;
+  value1 += 1;
+  value2 += 2;
+  value3 += 3;
+  value4 += 4;
+  value5 += 5;
 
   hibike_loop();
 }
 
 
-// you must implement this function. It is called when the device receives a DeviceUpdate packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_update(uint8_t param, uint32_t value) {
-  if (param < NUM_PARAMS) {
-    params[param] = value;
-    return params[param];
-  }
-  return ~((uint32_t) 0);
-}
+// You must implement this function.
+// It is called when the device receives a Device Write packet.
+// Updates param to new value passed in data.
+//    param   -   Parameter index
+//    data    -   value to write, in little-endian bytes
+//    len     -   number of bytes in data
+//
+//   return  -   size of bytes written on success; otherwise return 0
 
-// you must implement this function. It is called when the devie receives a DeviceStatus packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_status(uint8_t param) {
-  if (param < NUM_PARAMS) {
-    return params[param];
+uint32_t device_write(uint8_t param, uint8_t* data, size_t len) {
```

Comment:
Same issue as before; len is a buffer size and should not be the argument to `read_num_bytes()`